intelmq/bots/experts/filter/expert.py

```diff
@@ -1,22 +1,78 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
+
+import re
 import sys
+import pytz
+from datetime import datetime, timedelta
 
+from dateutil import parser
 from intelmq.lib.bot import Bot
+from intelmq.lib.harmonization import DateTime
+
 
+class FilterExpertBot(Bot):
 
-class FilterBot(Bot):
+    # number of minutes in time units
+    timespans = ({'hour': 60, 'day': 24*60, 'week': 7*24*60, 'month': 30*24*60, 'year': 365*24*60})
+
+    # parse relative time attributes
+    @staticmethod
+    def parse_relative(relative_time):
+        try:
+            result = re.findall(r'(\d+)\s+(\w+[^s])s?', relative_time, re.UNICODE)
```

Comment:
`'-2 months'` matches and becomes `[('2', 'month')]`. Add markers for begin and end of string: `r'^(\d+)\s+(\w+[^s])s?$'`.